internal/database/badger/database.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2022 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+// Package badger provides a database implementation using badger v3.
+package badger
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/internal/database"
+	"github.com/dgraph-io/badger/v3"
```

Comment:
nit: it's not obvious the package name in this folder is still `badger`
```suggestion
	badger "github.com/dgraph-io/badger/v3"
```

Reply:
Done